esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2298,6 +2264,66 @@ bool CFileSprayEx::onSprayVariable(IEspContext &context, IEspSprayVariable &req,
     return true;
 }
 
+void CFileSprayEx::checkDZScopeAccessAndSetSpraySourceDFUFileSpec(IEspContext &context, const char *srcPlane, const char *srcHost,
+    const char *srcFile, MemoryBuffer &srcXML, IDFUfileSpec *srcDFUfileSpec)
+{
+    if(srcXML.length() == 0)
+    {
+        RemoteMultiFilename rmfn;
+        SocketEndpoint ep(srcHost);
+        if (ep.isNull())
```

Comment:
question: what if there is no srcHost passed, only the srcPlane?

Which should be the way forward, both from fileservices _and_ from the UI.

Reply:
Should use the host read from previous code.

Reply:
The code here is in checkDZScopeAccessAndSetSpraySourceDFUFileSpec(). The code calls the readAndCheckSpraySourceReq() before calling checkDZScopeAccessAndSetSpraySourceDFUFileSpec(). Both sourcePlane and srcHost are already solved inside the readAndCheckSpraySourceReq().